test/Microsoft.AspNet.FileProviders.Combined.Tests/TestUtility/MockFileProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.FileProviders.Combined.Tests.TestUtility
+{
+    internal class MockFileProvider : IFileProvider
+    {
+        private IEnumerable<IFileInfo> _files;
+        private Dictionary<string, IChangeToken> _changeTokens;
+
+        public MockFileProvider()
+        {}
+
+        public MockFileProvider(params IFileInfo[] files)
+        {
+            _files = files;
+        }
+
+        public MockFileProvider(params KeyValuePair<string, IChangeToken>[] changeTokens)
+        {
+            _changeTokens = changeTokens.ToDictionary(changeToken => changeToken.Key,
+                                                      changeToken => changeToken.Value);
+        }
+
+        public IDirectoryContents GetDirectoryContents(string subpath)
+        {
+            if (string.IsNullOrEmpty(subpath))
+            {
+                return new EnumerableDirectoryContents(_files);
+            }
+
+            var filesInFolder = _files.Where(f => f.Name.StartsWith(subpath, StringComparison.Ordinal));
+            if (filesInFolder.Any())
+            {
+                return new EnumerableDirectoryContents(filesInFolder);
+            }
+            return new NotFoundDirectoryContents();
+        }
+
+        public IFileInfo GetFileInfo(string subpath)
+        {
+            var file = _files.FirstOrDefault(f => f.Name == subpath);
+            return file ?? new NotFoundFileInfo(subpath);
+        }
+
+        public IChangeToken Watch(string filter)
+        {
+            if(_changeTokens != null && _changeTokens.ContainsKey(filter))
```

Comment:
I think it would be fine to null ref if we called watch without registering any changeTokens.

```C#
if (_changeTokens.TryGetValue(filter, out changeToken))
{
   return changeToken;
}
```